src/components/Forms/FormHelperText/FormHelperText.tsx

```diff
@@ -1,46 +1,64 @@
 /* External dependencies */
-import React, { forwardRef, useMemo } from 'react'
+import React, { forwardRef, useCallback } from 'react'
 import { isEmpty } from 'lodash-es'
 
 /* Internal dependencies */
-import { Typography, SemanticNames } from 'Foundation'
+import { Typography } from 'Foundation'
+import useMergeRefs from 'Hooks/useMergeRefs'
+import useFormControlContext from 'Components/Forms/useFormControlContext'
 import type FormHelperTextProps from './FormHelperText.types'
 import * as Styled from './FormHelperText.styled'
 
 export const FORM_HELPER_TEXT_TEST_ID = 'bezier-react-form-helper-text'
 
-function FormHelperText({
-  id,
-  testId = FORM_HELPER_TEXT_TEST_ID,
-  as = 'p',
-  hasError,
-  children,
-  ...rest
-}: FormHelperTextProps,
-forwardedRef: React.Ref<HTMLParamElement>,
+function FormHelperText(
+  props: FormHelperTextProps,
+  forwardedRef: React.Ref<HTMLParamElement>,
 ) {
-  const color = useMemo<SemanticNames>(() => {
-    if (hasError) {
-      return 'bgtxt-orange-normal'
-    }
-    return 'txt-black-dark'
-  }, [hasError])
+  const {
+    testId = FORM_HELPER_TEXT_TEST_ID,
+    as = 'p',
+    typo = Typography.Size13,
+    children,
+    ...rest
+  } = props
+
+  const contextValue = useFormControlContext()
+
+  const {
+    hasError,
+    setHasHelperText,
+    Wrapper,
+    ...ownProps
+  } = contextValue?.getHelperTextProps(rest) ?? {
+    ...rest,
+    setHasHelperText: undefined,
+    Wrapper: React.Fragment,
+  }
+
+  const setHelperTextRef = useCallback((node: HTMLElement) => {
+    if (!node) { return }
+    setHasHelperText?.(true)
+  }, [setHasHelperText])
```

Comment:
```suggestion
  const setHelperTextRef = useCallback((node: HTMLElement) => {
    if (!node) { setHelperText?.(false) }
    else { setHasHelperText?.(true) }
  }, [setHasHelperText])
```

false로 바꾸는 부분이 없어서 체크해주세용

그리고 이렇게 ref로 여러 컴포넌트에서 체크하는 것보다, 그냥 FormControl에서 자기 children을 순회하면서 파악하는 편이 간결하지 않을까 하는 의견 드립니다. (desk의 FlatTable처럼)